pyiron_base/archiving/import_archive.py

```diff
@@ -0,0 +1,111 @@
+import os
+import pandas
+import numpy as np
+from shutil import rmtree
+from distutils.dir_util import copy_tree
+import tarfile
+from pyiron_base.generic.util import static_isinstance
+
+
+def getdir(path):
+    path_base_name = os.path.basename(path)
+    if path_base_name == "":
+        return os.path.basename(os.path.dirname(path))
+    else:
+        return path_base_name
+
+
+def update_id_lst(record_lst, job_id_lst):
+    masterid_lst = []
+    for masterid in record_lst:
+        if np.isnan(masterid):
+            masterid_lst.append(masterid)
+        elif isinstance(masterid, int) or isinstance(masterid, float):
+            masterid = int(masterid)
+            masterid_lst.append(job_id_lst[masterid])
+    return masterid_lst
+
+
+def extract_archive(archive_directory):
+    fname = archive_directory+".tar.gz"
+    tar = tarfile.open(fname, "r:gz")
+    tar.extractall()
+    tar.close()
+
+
+def import_jobs(
+    project_instance, directory_to_import_to, archive_directory,
+    df, compressed=True
+):
+    # Copy HDF5 files
+    # if the archive_directory is a path(string)/name of the compressed file
+    if static_isinstance(
+        obj=archive_directory,
+        obj_type='builtins.str'
+    ):
```

Comment:
```suggestion
    if isinstance(archive_directory, str):
```

Reply:
Sorry, this weekend I could not work. Thanks for updating it yourself.